tests/tests_pytorch/strategies/test_colossalai_strategy.py

```diff
@@ -0,0 +1,263 @@
+import os
+from typing import Any, Dict
+
+import pytest
+import torch
+import torch.nn.functional as F
+from torch import nn, Tensor
+from torch.optim import Optimizer
+from torchmetrics import Accuracy
+
+from pytorch_lightning import LightningModule, seed_everything, Trainer
+from pytorch_lightning.demos.boring_classes import BoringModel
+from pytorch_lightning.plugins.precision import ColossalAIPrecisionPlugin
+from pytorch_lightning.strategies import ColossalAIStrategy
+from pytorch_lightning.strategies.colossalai import _COLOSSALAI_AVAILABLE
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests_pytorch.helpers.datamodules import ClassifDataModule
+from tests_pytorch.helpers.runif import RunIf
+
+if _COLOSSALAI_AVAILABLE:
+    from colossalai.nn.optimizer import HybridAdam
+
+
+def test_invalid_colosalai(monkeypatch):
+    import pytorch_lightning.strategies.colossalai as colossal_strategy
+
+    monkeypatch.setattr(colossal_strategy, "_COLOSSALAI_AVAILABLE", False)
+    with pytest.raises(MisconfigurationException):
```

Comment:
Let's add the match= here for the message?

Reply:
Additional message is added.